utils/kubernetes/client.go

```diff
@@ -32,40 +33,71 @@ func DetectKubeConfig(configfile []byte) (config *rest.Config, err error) {
 	// Look for kubeconfig from the path mentioned in $KUBECONFIG
 	kubeconfig := os.Getenv("KUBECONFIG")
 	if kubeconfig != "" {
-		if config, err = clientcmd.BuildConfigFromFlags("", kubeconfig); err == nil {
+		_, cfgFile, err := ProcessConfig(kubeconfig, "")
+		if err != nil {
+			return nil, err
+		}
+		if config, err = clientcmd.RESTConfigFromKubeConfig(cfgFile); err == nil {
 			return config, err
 		}
 	}
 
 	// Look for kubeconfig at the default path
 	path := filepath.Join(utils.GetHome(), ".kube", "config")
-	if config, err = clientcmd.BuildConfigFromFlags("", path); err == nil {
+	_, cfgFile, err := ProcessConfig(path, "")
+	if err != nil{
+		return nil, err
+	}
+	if config, err = clientcmd.RESTConfigFromKubeConfig(cfgFile); err == nil {
 		return config, err
```

Comment:
```suggestion
		return config, err
```

While this is not immediately due to your changes, while you are here you can improve the logic to ensure meshkit error is returned.